src/report.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2017 Kisio Digital and/or its affiliates.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU Affero General Public License as published by the
+// Free Software Foundation, version 3.
+
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+// details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>
+
+//! Helpers to create a report for faillible processes.
+use serde::Serialize;
+
+/// Each report record will be categorized with a type implementing this
+/// `ReportType` trait.
+pub trait ReportType: Serialize + PartialEq {}
```

Comment:
Name the trait with a verb instead of a name.

At the very least `ReportCategory` is better (`Type` is connoted in Rust).

Reply:
Done in 7e0410ef902f40d3d1c835bf163b1b0d8609ca57.